frontend/src/components/Signup.js

```diff
@@ -11,61 +11,61 @@ function Signup() {
 
   const [isOpen, setIsOpen] = useState(false);
 
-  
-  function isProperSignup(...fields){
-	  //Note that weight is the last index in fields
-	  for(let i = 0; i < 4; i++)
-		  if(fields[i].length <= 0)
-			  errors.push(i);
-		  
-	  if(fields[3] != fields[4])
-		  errors.push(4);
-	  
-	  if(isNaN(fields[5]) || fields[5].length == 0)
-		  errors.push(5);
-	  
-	  return errors.length == 0;
-  }
-  
-  function addErrors(...fields){
-	  for(let i = 0; i < 4; i++)
-		  if(errors.includes(i)){
-			 //Add red warning to fields[i]
-		  }
-		 
-	  if(errors.includes(4)){
-		  //Add red warning to fields[4]
-	  }
-	  
-	  if(errors.includes(5)){
-		  //Add red warning to fields[5]
-	  }
-		  
+  function isProperSignup(...fields) {
+    //Note that weight is the last index in fields
+    for (let i = 0; i < 4; i++) if (fields[i].length <= 0) errors.push(i);
+
+    if (fields[3] != fields[4]) errors.push(4);
+
+    if (isNaN(fields[5]) || fields[5].length == 0) errors.push(5);
+
+    return errors.length == 0;
   }
-  
-  function addDuplicateEmailWarning(){
-	  
-	  
+
+  function addErrors(...fields) {
+    for (let i = 0; i < 4; i++)
+      if (errors.includes(i)) {
+        //Add red warning to fields[i]
+      }
+
+    if (errors.includes(4)) {
+      //Add red warning to fields[4]
+    }
+
+    if (errors.includes(5)) {
+      //Add red warning to fields[5]
+    }
   }
 
+  function addDuplicateEmailWarning() {}
+
   const doSignup = async (event) => {
     //event.preventDefault();
-
-	  errors = [];
-	
-    let fName = document.getElementById('fName');
-    let lName = document.getElementById('lName');
-    let weight = document.getElementById('weight');
-    let email = document.getElementById('email');
-    let password = document.getElementById('password');
-    let repeat = document.getElementById('repeat');
-	
-    if(!isProperSignup(fName.value, lName.value, email.value, password.value, repeat.value, weight.value)){
+    console.log("Hello");
+    errors = [];
+
+    let fName = document.getElementById("fName");
+    let lName = document.getElementById("lName");
+    let weight = document.getElementById("weight");
+    let email = document.getElementById("email");
+    let password = document.getElementById("password");
+    let repeat = document.getElementById("repeat");
+
+    if (
+      !isProperSignup(
+        fName.value,
+        lName.value,
+        email.value,
+        password.value,
+        repeat.value,
+        weight.value
+      )
+    ) {
```

Comment:
This is the old JS way of doing things, you should implement this with UseState. GetElementBy should not really exist in this project, maybe in another way but not like this.

Reply:
Noah made this, i'll let him know so that he is aware of it

Reply:
Did you mean useRef()? Or you mean something like " const[fName, setFName] = useState(getElementById("fName")); "

Reply:
" const[fName, setFName] = useState(document.getElementById("fName")); "